Jellyfin.Api/Controllers/DynamicHlsController.cs

```diff
@@ -1653,11 +1664,38 @@ private string GetCommandLineArguments(string outputPath, StreamState state, boo
                 segmentFormat,
                 startNumber.ToString(CultureInfo.InvariantCulture),
                 baseUrlParam,
-                isEventPlaylist ? "event" : "vod",
                 outputTsArg,
+                hlsArguments,
                 outputPath).Trim();
         }
 
+        /// <summary>
+        /// Gets the HLS arguments for transcoding.
+        /// </summary>
+        /// <returns>The command line arguments for HLS transcoding.</returns>
+        private string GetHlsArguments(bool isEventPlaylist, int segmentLength)
+        {
+            var enableThrottling = _encodingOptions.EnableThrottling;
+            var enableSegmentDeletion = _encodingOptions.EnableSegmentDeletion;
+
+            // Only enable segment deletion when throttling is enabled
+            if (enableThrottling && enableSegmentDeletion)
+            {
+                // Store enough segments for configured seconds of playback; this needs to be above throttling settings
+                var segmentCount = _encodingOptions.SegmentKeepSeconds / segmentLength;
+
+                _logger.LogDebug("Using throttling and segment deletion, keeping {0} segments", segmentCount);
+
+                return string.Format(CultureInfo.InvariantCulture, "-hls_list_size {0} -segment_wrap {0} -hls_flags delete_segments", segmentCount.ToString(CultureInfo.InvariantCulture));
```

Comment:
We typically use `hls_playlist_type vod`, which "Forces hls_list_size to 0; the playlist must not change." (https://ffmpeg.org/ffmpeg-formats.html)

Reply:
Without the vod type, you cannot seek ahead of the playlist, so I'm not sure how useful this is.

Reply:
> Without the vod type, you cannot seek ahead of the playlist, so I'm not sure how useful this is.

Seeking works perfectly fine. Obviously just within ~3 or so minutes since the throttler is on and outside of it a new transcoder instance is started.